pkg/apis/serving/v1/revision_lifecycle.go

```diff
@@ -184,7 +184,8 @@ func (rs *RevisionStatus) PropagateAutoscalerStatus(ps *autoscalingv1alpha1.PodA
 		rs.DesiredReplicas = ps.DesiredScale
 	}
 
-	if cond == nil {
+	sksCondition := ps.GetCondition(autoscalingv1alpha1.PodAutoscalerConditionSKSReady)
+	if cond == nil || (cond.IsUnknown() && sksCondition != nil && sksCondition.IsUnknown()) {
```

Comment:
Can you explain why we need the extra clauses? 

I would think we could simply have  `cond == nil || cond.IsUnknown()`

Reply:
I believe that there were cases now that the pa has conditions at the start, it was not waiting for the service to get ready, so I was trying to make sure to simulate the previous cases before defaulting pa conditions